tests/main/system-usernames-illegal/task.yaml

```diff
@@ -5,11 +5,15 @@ summary: ensure unapproved user cannot be used with system-usernames
 # we can ignore these.
 systems: [-amazon-linux-2-*, -centos-7-*, -debian-9-*, -debian-sid-*, -fedora-29-*, -fedora-30-*, -opensuse-15.0-*, -opensuse-15.1-*, -ubuntu-14.04-*]
 
-restore:
+restore: |
     # Make sure the snap is removed if the test failed and the snap was
     # installed
     snap remove test-snapd-illegal-system-username || true
 
+    # snapd will create this for us, but we'll remove it for consistency in
+    # test runs
+    user-tool remove-with-group snap_daemon
```

Comment:
I think we should tweak this a bit, instead of removing the snap_daemon user we should add code to check that *no* snap_daemon was created. So something like:
```
not getent passwd snap_daemon
not getent group snap_daemon
```
We do not want an invalid snap to create users.